src/V3PerPackage/PerBatchProcessor.cs

```diff
@@ -56,6 +67,39 @@ public class PerBatchProcessor
             return true;
         }
 
+        private async Task<IReadOnlyList<PerPackageContext>> GetExistingPackagesAsync(IReadOnlyList<PerPackageContext> packageContexts)
+        {
+            var tasks = packageContexts
+                .Select(x => new { Context = x, Task = DoesPackageExistAsync(x) })
+                .ToList();
+
+            await Task.WhenAll(tasks.Select(x => x.Task));
+
+            return tasks
+                .Where(x => x.Task.Result)
+                .Select(x => x.Context)
+                .ToList();
+        }
+
+        private async Task<bool> DoesPackageExistAsync(PerPackageContext packageContext)
+        {
+            using (var request = new HttpRequestMessage(HttpMethod.Head, packageContext.PackageUri))
```

Comment:
Note that this skips packages that exist in V2 but for whatever reason are missing from the packages container. I think it might be useful to add that in a comment.

Reply:
Sure, sounds good.

Reply:
Done.